app/javascript/components/common/ComboButton.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useContext } from 'react'
+import ReactDOM from 'react-dom'
+import { usePanel } from '../../hooks/use-panel'
+import { GraphicalIcon } from './GraphicalIcon'
+
+const ComboButtonContext = React.createContext({
+  open: false,
+  setPanelElement: (
+    value: React.SetStateAction<HTMLDivElement | null>
+  ): void => {},
+  styles: {},
+  attributes: {},
+})
+
+export const ComboButton = ({
+  className = '',
+  children,
+}: React.PropsWithChildren<{ className?: string }>): JSX.Element => {
+  const {
+    open,
+    setOpen,
+    setButtonElement,
+    setPanelElement,
+    styles,
+    attributes,
+  } = usePanel({
+    placement: 'bottom-end',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 14],
+        },
+      },
+    ],
+  })
+
+  const classNames = ['c-combo-button', className]
+
+  return (
+    <ComboButtonContext.Provider
+      value={{
+        setPanelElement: setPanelElement,
+        open: open,
+        styles: styles,
+        attributes: attributes,
+      }}
+    >
+      <div className={classNames.join(' ')}>
+        {children}
+        <div className="--dropdown-segment">
+          <button
+            onClick={() => {
+              setOpen(!open)
+            }}
+            ref={setButtonElement}
+          >
+            <GraphicalIcon icon="chevron-down" />
+          </button>
+        </div>
+      </div>
+    </ComboButtonContext.Provider>
+  )
+}
+
+ComboButton.EditorSegment = ({ children }: React.PropsWithChildren<{}>) => {
+  return <div className="--editor-segment">{children}</div>
+}
```

Comment:
This was renamed to `--primary-segement` on `main` btw (I didn't get the merge in of that PR last night before yu started - sorry!)

Reply:
No problem :)